models/discordactions.js

```diff
@@ -831,6 +840,175 @@ const updateUsersWith31DaysPlusOnboarding = async () => {
   }
 };
 
+const getMissedProgressUpdatesUsers = async (options = {}) => {
+  const { cursor, size = 500, excludedDates = [], excludedDays = [0], dateGap = 3 } = options;
+  const stats = {
+    tasks: 0,
+    missedUpdatesTasks: 0,
+  };
+  try {
+    const discordUsersPromise = discordService.getDiscordMembers();
+    const missedUpdatesRoleId = discordMissedUpdatesRoleId;
+
+    let gapWindowStart = Date.now() - convertDaysToMilliseconds(dateGap);
+    const gapWindowEnd = Date.now();
+    excludedDates.forEach((timestamp) => {
+      if (timestamp > gapWindowStart && timestamp < gapWindowEnd) {
+        gapWindowStart -= convertDaysToMilliseconds(1);
+      }
+    });
+
+    if (excludedDays.length === 7) {
+      return { usersToAddRole: [], ...stats };
+    }
+
+    for (let i = gapWindowEnd; i >= gapWindowStart; i -= convertDaysToMilliseconds(1)) {
+      const day = new Date(i).getDay();
+      if (excludedDays.includes(day)) {
+        gapWindowStart -= convertDaysToMilliseconds(1);
+      }
+    }
+
+    let taskQuery = buildTasksQueryForMissedUpdates(gapWindowStart, size);
+
+    if (cursor) {
+      const data = await tasksModel.doc(cursor).get();
+      if (!data.data()) {
+        return {
+          statusCode: 400,
+          error: "Bad Request",
+          message: `Invalid cursor: ${cursor}`,
+        };
```

Comment:
Instead of passing a different field back and adding an extra check in controller file, decided to pass the error message, had also seen the same being done in other places in our repo. 

- Let me know if there are other good and simplified approach.